xbmc/guilib/GUIWindowManager.cpp

```diff
@@ -810,6 +813,16 @@ void CGUIWindowManager::CloseDialogs(bool forceClose) const
   }
 }
 
+void CGUIWindowManager::CloseModalDialogs(bool forceClose) const
+{
+  CSingleLock lock(g_graphicsContext);
```

Comment:
this looks deadlock happy. i won't hold the lock for expensive operations like closing dialogs. why is this needed?